lib/net/packets.js

```diff
@@ -957,6 +971,681 @@ class NotFoundPacket extends InvPacket {
+  /**
+   * Create a `getcfheaders` packet.
+   * @constructor
+   * @param {(Hash[])?} locator
+   * @param {Hash?} stop hash
+   * @param {Number} filter type
+   */
+
+  constructor(locator, stop, filterType) {
+    super();
+
+    this.cmd = 'getcfheaders';
+    this.type = exports.types.GETCFHEADERS;
+
+    this.locator = locator || [];
+    this.stop = stop || null;
+    this.filterType = filterType;
+  }
+
+  /**
+   * Get serialization size.
+   * @returns {Number}
+   */
+
+  getSize() {
+    let size = 0;
+    size += encoding.sizeVarint(this.locator.length);
+    size += 32 * this.locator.length;
+    size += 33;
+    return size;
+  }
+
+  /**
+   * Serialize getcfheaders packet to writer.
+   * @param {BufferWriter} bw
+   */
+
+  toWriter(bw) {
+    assert(this.locator.length <= 50000, 'Too many block hashes.');
+
+    bw.writeVarint(this.locator.length);
+
+    for (const hash of this.locator)
+      bw.writeHash(hash);
+
+    bw.writeHash(this.stop || consensus.ZERO_HASH);
+    bw.writeU8(this.filterType);
+
+    return bw;
+  }
+
+  /**
+   * Serialize getcfheaders packet.
+   * @returns {Buffer}
+   */
+
+  toRaw() {
+    const size = this.getSize();
+    return this.toWriter(bio.write(size)).render();
+  }
+
+  /**
+   * Inject properties from buffer reader.
+   * @private
+   * @param {BufferReader} br
+   */
+
+  fromReader(br) {
+    const count = br.readVarint();
+
+    assert(count <= 50000, 'Too many block hashes.');
+
+    for (let i = 0; i < count; i++)
+      this.locator.push(br.readHash('hex'));
+
+    this.stop = br.readHash('hex');
+
+    if (this.stop === encoding.NULL_HASH)
+      this.stop = null;
+
+    this.filterType = br.readU8();
+
+    return this;
+  }
+
+  /**
+   * Inject properties from serialized data.
+   * @private
+   * @param {Buffer} data
+   */
+
+  fromRaw(data) {
+    return this.fromReader(bio.read(data));
+  }
+
+  /**
+   * Instantiate getcfheaders packet from buffer reader.
+   * @param {BufferReader} br
+   * @returns {GetCFHeadersPacket}
+   */
+
+  static fromReader(br) {
+    return new this().fromRaw(br);
+  }
+
+  /**
+   * Instantiate getcfheaders packet from serialized data.
+   * @param {Buffer} data
+   * @param {String?} enc
+   * @returns {GetCFHeadersPacket}
+   */
+
+  static fromRaw(data, enc) {
+    if (typeof data === 'string')
+      data = Buffer.from(data, enc);
+    return new this().fromRaw(data);
+  }
+}
+
+/**
+ * CFHeaders Packet
+ * @extends Packet
+ * @property {Hash} stop hash
+ * @property {Number} filter type
+ * @property {Hash[]} header hashes
+ */
+
+class CFHeadersPacket extends Packet {
+  /**
+   * Create a `cfheaders` packet.
+   * @constructor
+   * @param {Hash?} stop hash
+   * @param {Number} filter type
+   * @param {(Hash[])?} header hashes
+   */
+
+  constructor(stop, filterType, hashes) {
+    super();
+
+    this.cmd = 'cfheaders';
+    this.type = exports.types.CFHEADERS;
+
+    this.stop = stop || null;
```

Comment:
Why is the fallback `null` instead of `encoding.ZERO_HASH`? Other packets usually use the same type but a null value of that type for the fallback value